index/templates/index/index.html

```diff
@@ -118,7 +121,16 @@ <h3 style="display: inline;"><b>Announcements</b></h3>
               <a href="{% url 'contest:contest' c_upcoming.id %}">{{ c_upcoming.cname }}</a>
```

Comment:
Please remove the link when request user can not view the contest. (only admin and contest owner / coowner can view it)